### PR TITLE
dsv3 with compute_additional_head() + classifier config

### PR DESCRIPTION
**Pull request by erickawu**
- mixin class to add classifier heads + config. should be extensible to qwen
- to test, run w ds weights overridden with `{"architectures": ["DeepseekV3ForCausalLMWithAdditionalHeads"]}` + classifiers in my nb1 home dir. non-streaming completions request will return
`
{"id":"cmpl-9bef68165d824dd9963d18688dae2113","object":"text_completion","created":1754495706,"model":"deepseekv3","choices":[{"index":0,"text":" city that is known for its vibrant culture, diverse population","logprobs":null,"finish_reason":"length","stop_reason":null,"prompt_logprobs":null,"additional_heads":[[0.50390625,0.5234375,0.50390625],[0.515625,0.515625,0.50390625],[0.4921875,0.5,0.50390625],[0.49609375,0.5078125,0.5078125],[0.5234375,0.51171875,0.5078125],[0.4765625,0.515625,0.5078125],[0.4453125,0.515625,0.5078125],[0.4609375,0.5234375,0.5078125],[0.46484375,0.5078125,0.50390625],[0.462890625,0.51171875,0.5078125],[0.48046875,0.51953125,0.50390625]],"is_filtered":false}],"usage":{"prompt_tokens":5,"total_tokens":16,"completion_tokens":11,"prompt_tokens_details":null},"kv_transfer_params":null}
`

### CODE REVIEW

File: vllm/model_executor/layers/additional_heads.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                "Ensure that the parent class is compatible with WithAdditionalHeads."
+            )
+
+        loaded_params = original_cls.load_weights(weights)
+        head_loaded_params = self._load_classification_heads_from_files()
+        logger.info(f"Loaded {len(head_loaded_params)} additional head parameters.")
+        return loaded_params.union(head_loaded_params)
+
+    def compute_additional_head(
+        self, hidden_states: torch.Tensor, additional_heads_extra_inputs: Optional[List[Dict[str, Any]]] = None
```
[the remark this answers is not in the document]
Comment (erickawu) [reply vllm/model_executor/layers/additional_heads.py:124]: just needs to be specified as input to run, method expects 3 args

File: vllm/model_executor/layers/additional_heads.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            raise ValueError("additional_heads_config is required in additional_config")
+    
+        return [
+            ClassifierConfig.from_dict(config) for config in additional_heads_config
+        ]
+    
+    def _build_classification_head(self, num_hidden_layers: int, hidden_dim: Optional[int]) -> nn.Sequential:
+        """Build a binary classification head with the specified number of hidden layers."""        
+        if num_hidden_layers == 0:
+            return nn.Sequential(nn.Linear(self.config.hidden_size, 1), nn.Sigmoid())
```
[the remark this answers is not in the document]
Comment (erickawu) [reply vllm/model_executor/layers/additional_heads.py:82]: okkk ill remove the sigmoid
